quic/s2n-quic-core/src/event/metrics/aggregate/variant.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::info::{Str, Variant};
+
+pub trait AsVariant {
+    const VARIANTS: &'static [Variant];
+
+    fn variant_idx(&self) -> usize;
+
+    #[inline]
+    fn as_variant(&self) -> &'static Variant {
+        &Self::VARIANTS[self.variant_idx()]
+    }
+}
+
+impl AsVariant for bool {
```

Comment:
Unclear, but it's worth noting bool might be special -- the utility of e.g. `avg` metrics for it are much more likely to be useful than typically found for enum values in general. A dedicated BoolCounter could make sense.

Reply:
That's fair. I can add a `BoolCounter` instead.